graphql_compiler/compiler/subclass.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2017-present Kensho Technologies, LLC.
+from graphql.type.definition import GraphQLObjectType, GraphQLUnionType
+import six
+
+
+def _transitive_closure(graph):
+    """Compute the transitive closure of a graph represented as dict inplace."""
+    # Floyd-Warshall O(N^3)
+    for k, k_out in six.iteritems(graph):
+        for i, i_out in six.iteritems(graph):
+            for j, j_out in six.iteritems(graph):
+                if k in i_out and j in k_out:
+                    graph.setdefault(i, set()).add(j)
```

Comment:
Because of the "class is a subclass of itself" rule, I think the `setdefault` here is unnecessary.

Reply:
But it's not specified in the spec of this function. I'd rather not complicate it.

Reply:
I think it's better to specify it. Otherwise, this line is mutating the keys of a dict that is being iterated over, which is definitely bad practice. With that in the spec, it's only mutating the values, which is somewhat more palatable.